superpixel_classification/SuperpixelClassification/SuperpixelClassificationTorch.py

```diff
@@ -133,11 +129,177 @@ def mc_forward_impl(self, input: torch.Tensor) -> torch.Tensor:
+        self.device: str = torch.device(
+            ('cuda' if torch.cuda.is_available() and torch.cuda.device_count() > 0 else 'cpu'),
+        )
+        # print(f'Initial model.device = {self.device}')
+        super(_BayesianVectorTorchModel, self).__init__()
+
+        self.input_dim: int = input_dim
+        self.num_classes: int = num_classes
+        self.bayesian_samples: int = 12
+        self.fc: torch.Module = torch.nn.Linear(input_dim, num_classes)
+        self.fc_drop: torch.Module = bbald.consistent_mc_dropout.ConsistentMCDropout()
+
+    def mc_forward_impl(self, input: torch.Tensor) -> torch.Tensor:
+        # TODO: Is torch.mul appropriate here?
+        input = torch.mul(input, 1.0 / 255)
+        input = self.fc(input)
+        # TODO: Is it appropriate to have fc_drop as a last layer; we don't do that for
+        # batchbald on patches?  More generally, is subclassing from bbald and using
+        # self.bayesian_samples during training and/or prediction sufficient to make
+        # this model properly Bayesian?
+        input = self.fc_drop(input)
+        # To remain consistent with the Tensorflow implementation, we will not include
+        # `input = torch.nn.functional.log_softmax(input, dim=1)` at this point.
+        return input
+
+
 class SuperpixelClassificationTorch(SuperpixelClassificationBase):
     def __init__(self):
         self.training_optimal_batchsize: Optional[int] = None
         self.prediction_optimal_batchsize: Optional[int] = None
 
+    def initializeCreateFeatureFromPatchAndMaskUNI(self):
+        import timm
+        import timm.data
+        import timm.data.transforms_factory
+
+        """
+        To make the timm.create_model call succeed, be sure that a command like the following has
+        already been run from a bash prompt on each system supporting the
+        dsarchive/superpixel:latest docker image.  We need to run this command only if
+        .cache/huggingface/hub does not already have the MahmoodLab/UNI model.  Instead of
+        $HOME/.cache/huggingface/token in the following, use the actual location of your HuggingFace
+        token; for security reasons it is better if the token is not within the mount point, which
+        in this example is the tree rooted from the directory $HOME/.cache/huggingface/hub.
+
+          docker run \
+            --rm \
+            --env=HF_TOKEN=$(cat $HOME/.cache/huggingface/token) \
+            -v $HOME/.cache/huggingface/hub:/root/.cache/huggingface/hub:rw \
+            --entrypoint "" \
+            dsarchive/superpixel:latest \
+            huggingface-cli download MahmoodLab/UNI
+
+        Additionally, make sure that your `docker-compose.override.yml` file includes something like
+
+          services:
+            worker:
+              environment:
+                GIRDER_WORKER_DOCKER_RUN_OPTIONS: >-
+                  {"volumes": ["/path_to_my_home_directory/.cache/huggingface/hub:/root/.cache/huggingface/hub:ro"],
```

Comment:
I've found the only way to make the line length check happy is to add `# noqa` at the end of the line...

Reply:
We are now using YAML's `>-`, but that seems to work only if we want line breaks and subsequent whitespace replaced with a single space rather than with nothing.  So, the example is now:
```yaml
services:
  worker:
    environment:
      GIRDER_WORKER_DOCKER_RUN_OPTIONS: >-
        {"volumes":
          ["/path_to_home_directory/.cache/huggingface/hub:/root/.cache/huggingface/hub"],
          "environment": {"HF_HUB_OFFLINE": "1"}}
```